### ISSUE TITLE
intl_example tests do not run and documentation is scarce

### ISSUE THREAD

**Issue by TimotheosFirestone** [flutter/website#2546]
The `README.md` of https://github.com/flutter/website/tree/master/examples/internationalization/intl_example says that `flutter test` should be run, but the tests do not compile.

Changing the test main body to this compiles:

```dart
void main() {
  testWidgets('Smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(Demo());

    /* TODO: Implement working tests...

    // TODO: Set the language to English

    // Verify that our text is English
    expect(find.text('Hello World'), findsOneWidget);
    expect(find.text('Hola Mundo'), findsNothing);

    // TODO: Set the language to Spanish
    await tester.pump();

    // Verify that our text is Spanish
    expect(find.text('Hello World'), findsNothing);
    expect(find.text('Hola Mundo'), findsOneWidget);
    */
  });
}
```

but that's not a useful test, and I don't know how to complete the TODOs. It's very difficult to see the code in action and follow it, without being able to look for what happens when the language changes (or even know how to test that in my own app easily).

The documentation is also fairly scarce, especially for someone new to Flutter. It would be excellent if the `README.md` file contained an overview of internationalisation, such as how to organise the project and what to do to add another `Widget` with a new string. It would be excellent if the example code was commented, because it's hard to follow without any context, especially with a lot of the asynchronous code (why does it have to be asynchronous?).

**Comment by sfshaza2** [comment]
I'm surprised that the example doesn't work, as that should shake out in smoke testing. @HansMuller, any insight into this?
